sui_core/src/authority.rs

```diff
@@ -329,10 +330,13 @@ impl AuthorityState {
 
         // At this point we need to check if any shared objects need locks,
         // and whether they have them.
-        let has_shared_object = objects_by_kind
+        let shared_objects: Vec<_> = objects_by_kind
```

Comment:
nit: may be clearer to rename all `shared_objects` to `shared_object_refs`

Reply:
Good idea, did that!